pkg/core/consensus/agreement/step.go

```diff
@@ -105,7 +105,10 @@ func (s *Loop) shouldCollectNow(a message.Message, round uint64, queue *consensu
 		return false
 	}
 
-	if hdr.Round > round {
+	// Only store events up to 10 rounds into the future
+	// XXX: According to protocol specs, we should abandon consensus
+	// if we notice valid messages from far into the future.
+	if hdr.Round > round && hdr.Round-round < 10 {
```

Comment:
Out of curiosity, have we noticed a devnet node collecting an Agreement message from  so far round? 
Much earlier, we should have this node in `OutSync` mode and Consensus/Agreement goroutines terminated.

Reply:
Not yet, but this is possible in syncing/real network scenarios. Of course, there needs to be some proper guarding in place against maliciousness but it's part of the protocol to terminate and sync when noticing consensus messages from far in the future.